### PR TITLE
Support bumper sensors in Gazebo

### PR DESCRIPTION
**Pull request by sasilva1998**
# Description
- Added the support for bumper sensor for each of the iRobot Create versions. The contact with the bumpers is checked according to the mesh of the corresponding bumper shape for each robot. For that I added a new arg `version` in `create_description/urdf/create_base.urdf.xacro`.
- Unfortunately, since there is no plugin for Omni IR, could only add a the common IR from Gazebo which I hope is still of use.
- Generated a package with a minimal example to simulate the different Create robot version. Also updated the Readme to specify how to run the simulation. Uses a world file from `turtlebot3_gazebo`.
- Added the publish of transforms for the wheels.

This PR is connected to #76.

**Test configuration**
ROS version: Noetic

### CODE REVIEW

File: create_description/urdf/create_base_gazebo.urdf.xacro [hunk 1]
```diff
@@ -81,6 +82,63 @@
...
+          <bumperTopicName>bumper</bumperTopicName>
+          <frameName>bumper_link</frameName>
+        </plugin>
+      </sensor>
+    </gazebo>
+  </xacro:macro>
+
+  <xacro:macro name="sim_ir">
+    <gazebo reference="omni_ir_link">
+      <sensor type="ray" name="TeraRanger">
```
Comment (jacobperron) [inline create_description/urdf/create_base_gazebo.urdf.xacro]: Since the IR sensor on the Create is not a range finder, and there is no equivalent in Gazebo, I think we should just remove the simulated sensor from the URDF.

File: create_gazebo/launch/create_gazebo.launch [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    <arg name="headless" default="false" />
+    <arg name="paused" default="false" />
+    <arg name="use_sim_time" default="true" />
+
+    <arg name="rviz" default="false"/>
+
+    <arg name="x_pos" default="0" />
+    <arg name="y_pos" default="0" />
+
+    <arg name="world_name" default="$(find turtlebot3_gazebo)/worlds/turtlebot3_house.world"/>
```
Comment (jacobperron) [inline create_gazebo/launch/create_gazebo.launch]: I'd rather not add a dependency on the turtlebot3 packages. It seems like it adds a lot of installation overhead just to get a world file.

File: README.md [hunk 3]
```diff
@@ -4,9 +4,10 @@
 This package wraps the C++ library [libcreate][libcreate], which uses iRobot's [Open Interface Specification][oi_spec].
 
 <!--[](* Documentation: TODO)-->
-* ROS wiki page: http://wiki.ros.org/create_robot
-* Support: [ROS Answers (tag: create_robot)](http://answers.ros.org/questions/scope:all/sort:activity-desc/tags:create_robot/page:1/)
-* Author: [Jacob Perron](http://jacobperron.ca) ([Autonomy Lab](http://autonomylab.org), [Simon Fraser University](http://www.sfu.ca))
+
+- ROS wiki page: http://wiki.ros.org/create_robot
+- Support: [ROS Answers (tag: create_robot)](http://answers.ros.org/questions/scope:all/sort:activity-desc/tags:create_robot/page:1/)
+- Author: [Jacob Perron](http://jacobperron.ca) ([Autonomy Lab](http://autonomylab.org), [Simon Fraser University](http://www.sfu.ca))
```
Comment (jacobperron) [inline README.md]: Please revert this change and other whitespace/style changes below. They are not relevant to adding support for bumper and IR sensors.

File: README.md [hunk 4]
```diff
@@ -225,41 +237,58 @@ angular.z (+)     Rotate counter-clockwise (rad/s)
...
-``` bash
+```bash
 $ roslaunch create_bringup joy_teleop.launch [joy_config:=xbox360]
 ```
 
 There exists configuration files for the [Xbox 360 wired controller](https://www.amazon.ca/Microsoft-Xbox-360-Wired-Controller/dp/B003ZSN600) and the [Logitech F710 controller](http://gaming.logitech.com/en-ca/product/f710-wireless-gamepad). You can adapt these files for your preferred joystick configuration.
 
+## Simulation
+
+The package `create_gazebo` has a launch file with a minimal simulation example of the Create robots.
```
Comment (jacobperron) [inline README.md]: I have mixed feelings about adding a simulation-specific package. The package is not providing any custom simulation plugins or configuration; it is simply an example launch file and rviz config file. I think it makes more sense to simply move those into `create_bringup` (which is the package meant for example launch/config files). While I think having the examples are nice, it technically introduces a runtime dependency on Gazebo, which I'd rather avoid.

It might be best to keep these examples in a separate repository.

**Review by jacobperron — COMMENTED** [review]
Thanks for the additions!

See my feedback below.

**Comment by sasilva1998** [comment]
Hi @jacobperron, sorry for the late response and thanks for your comments. I have made the recommended changes.

**Review by jacobperron — APPROVED** [review]

**Comment by jacobperron** [comment]
Thanks for the contribution!
